### PR TITLE
Added toImage method

### PR DESCRIPTION
**Pull request by wolfenrain**
# Description

Please include a summary of the change.

Fixes # (issue number, if there is one)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

If something is unclear, please submit the PR anyways and ask about what you thought was unclear.

- [ ] This branch is based on the latest `master`
- [ ] I have added an entry under `[next]` in `CHANGELOG.md`
- [ ] I have formatted my code with `flutter format`
- [ ] I have made corresponding changes to the documentation
- [ ] I have added examples for new features in `doc/examples`
- [ ] The continuous integration (CI) is passing

### CODE REVIEW

File: lib/src/sprite.dart [hunk 1]
```diff
@@ -76,4 +78,11 @@ class Sprite {
 
     canvas.drawImageRect(image, src, drawRect, drawPaint);
   }
+
+  /// Return a new Image based on the [src] of the Sprite.
```
Comment (erickzanardo) [inline lib/src/sprite.dart:82]: Add a comment here to warn people about this being a a heavy operation and that it shouldn't be called on game render. What do you think?

File: lib/src/sprite.dart [hunk 2]
```diff
@@ -1,5 +1,7 @@
 import 'dart:ui';
 
+import 'package:flame/image_composition.dart';
```
Comment (luanpotter) [inline lib/src/sprite.dart:3]: https://github.com/luanpotter/vscode-dart-import :p

File: lib/src/sprite.dart (hunk 1 above)
Comment (wolfenrain) [reply lib/src/sprite.dart:82]: Added it!

File: lib/src/sprite.dart [hunk 3]
```diff
@@ -1,5 +1,7 @@
 import 'dart:ui';
 
+import '../image_composition.dart';
+
```
Comment (luanpotter) [inline lib/src/sprite.dart:4]: extra empty line :p 

the import blocks are flutter/ext libraries and relative imports (that are separated by empty lines)

**Review by luanpotter — APPROVED** [review]
LGTM

File: lib/image_composition.dart [hunk 4]
```diff
@@ -39,6 +39,9 @@ class _Composed {
 }
 
 /// The [ImageComposition] allows for composing multiple images onto a single image.
+/// 
+/// **Note:** Composing images is a heavy async operation and should not be called on each [Game.render].
+
```
Comment (luanpotter) [inline lib/image_composition.dart:44]: nit extra empty line (join doc to the class)

Comment (wolfenrain) [reply lib/image_composition.dart:44]: Woepsiedoepsie

**Pull request merged**